src/connector/builder.rs

```diff
@@ -335,5 +341,14 @@ mod tests {
             &connector.tls_config.alpn_protocols,
             &[b"h2".to_vec(), b"http/1.1".to_vec()]
         );
+        let connector = super::ConnectorBuilder::new()
```

Comment:
I think this test is only checking the result of using `enable_all_versions` with both the http1 and http2 feature flags enabled based on the guard on `test_alpn`. Ideally we could assert the correct state for more combinations than just this case.

Reply:
> Ideally we could assert the correct state for more combinations than just this case.

Yeah, I realized this too. I locally created an additional test case for only `http2` while I was writing the PR description. I suspected that I'd get feedback on the doc line so I didn't want to muddy the waters with an additional small commit. But I'll push it now.